sys/net/gnrc/routing/rpl/of0.c

```diff
@@ -79,11 +81,21 @@ uint16_t calc_rank(gnrc_rpl_parent_t *parent, uint16_t base_rank)
 /* We simply return the Parent with lower rank */
 gnrc_rpl_parent_t *which_parent(gnrc_rpl_parent_t *p1, gnrc_rpl_parent_t *p2)
 {
-    if (p1->rank <= p2->rank) {
-        return p1;
+    if (parent_cmp(p1, p2)) {
```

Comment:
I need to insist on `if (parent_cmp(p1, p2) > 0) {` because `-1` will also eval as true, only `0` is false